### PR TITLE
Add ToString and Print method for JoinRelationSetManager and Fix JoinNode Print

### PR DESCRIPTION
**Pull request by Light-City**
1.When we want to debug JoinRelationSetManager., we need to add the ToString and Print method.

for example: explain SELECT    COUNT(*)  FROM    t1, t2, t3, t4  WHERE    t1.c1 = t2.c1 AND    t2.c2 = t3.c2 AND    t3.c3 = t4.c3 ;

set_manager.Print();

will output:

```
child: 0
  relation: [0]
  child: 1
      relation: [0, 1]
child: 1
  relation: [1]
  child: 2
      relation: [1, 2]
child: 3
  relation: [3]
child: 2
  relation: [2]
  child: 3
      relation: [2, 3]
```

2.Change PrintJoinNode to Print and implement it.

### CODE REVIEW

**Review by Tmonster — CHANGES_REQUESTED** [review]
Thank you for the PR! This should indeed come in handy when debugging. Just have some notes. 

The RelationSetManager keeps track of already created relations. The way these references are stored isn't very intuitive, and muddies the logic as to what the RelationSetManager is actually doing, but it's an efficient implementation if joins have a very large number of relations. Currently it is a type of tree structure where every node has a reference to the relation set. 

That being said, can we instead just iterate over all the JoinRelationSets/Values in the tree, add each `JoinRelationSet` to a "to_print" variable and print all the join relations created so far? This way we don't have to do this indenting stuff with "child" nodes. Relations technically don't even have children.

For your example, I think a better output would look like this.

```
RelationSets created:
[0]
[0,1]
[1]
[1,2]
[3]
[2]
[2,3]
```

File: src/optimizer/join_order/join_relation_set.cpp [hunk 1]
```diff
@@ -139,4 +139,32 @@ JoinRelationSet &JoinRelationSetManager::Union(JoinRelationSet &left, JoinRelati
...
+	for (int i = 0; i < indent; ++i) {
+		result += "  ";
+	}
+	return result;
+}
+
+static string JoinRelationTreeNodeToString(const JoinRelationTreeNode *node, int indent) {
+	string result = "";
+	string indent_prefix = GetIndent(indent);
+	if (node->relation)
```
Comment (Tmonster) [inline src/optimizer/join_order/join_relation_set.cpp:153]: nit: can you wrap this in curly braces? Just to match the styling of the rest of the code base, we wrap all if statements in curly braces

Comment (Light-City) [reply src/optimizer/join_order/join_relation_set.cpp]: ok ,I changed.

**Comment by Light-City** [comment]
> Thank you for the PR! This should indeed come in handy when debugging. Just have some notes.
> 
> The RelationSetManager keeps track of already created relations. The way these references are stored isn't very intuitive, and muddies the logic as to what the RelationSetManager is actually doing, but it's an efficient implementation if joins have a very large number of relations. Currently it is a type of tree structure where every node has a reference to the relation set.
> 
> That being said, can we instead just iterate over all the JoinRelationSets/Values in the tree, add each `JoinRelationSet` to a "to_print" variable and print all the join relations created so far? This way we don't have to do this indenting stuff with "child" nodes. Relations technically don't even have children.
> 
> For your example, I think a better output would look like this.
> 
> ```
> RelationSets created:
> [0]
> [0,1]
> [1]
> [1,2]
> [3]
> [2]
> [2,3]
> ```

Indeed, I very much agree with your point of view. I have made modifications to the code and reviewed it diligently.

**Comment by Light-City** [comment]
In addition, I found that PrintJoinNode of JoinNode is not defined. Here I modify it to Print and implement it.

**Comment by Light-City** [comment]
> In addition, I found that PrintJoinNode of JoinNode is not defined. Here I modify it to Print and implement it.

related issue is https://github.com/duckdb/duckdb/issues/9067

**Review by Tmonster — APPROVED** [review]
Thanks! Pr looks good to me

**Comment by Light-City** [comment]
@Tmonster @Mytherin 

Can this PR be merged? Are there any modifications needed?

**Comment by Mytherin** [comment]
Thanks

**Pull request merged**